File: tests/0235_lowest_common_ancestor_of_a_binary_search_tree_test.cpp
#include "0235_lowest_common_ancestor_of_a_binary_search_tree.cpp"
#include <doctest.h>
#include <memory>

TEST_CASE("lowestCommonAncestor1")
{
    auto v = tree::make_tree({6, 2, 8, 0, 4, 7, 9, std::nullopt, std::nullopt, 3, 5});

    Solution s = Solution();
    CHECK(s.lowestCommonAncestor(v[0].get(), v[1].get(), v[2].get()) == v[0].get());
    CHECK(s.lowestCommonAncestor(v[0].get(), v[2].get(), v[1].get()) == v[0].get());

    CHECK(s.lowestCommonAncestor(v[0].get(), v[1].get(), v[4].get()) == v[1].get());
    CHECK(s.lowestCommonAncestor(v[0].get(), v[4].get(), v[1].get()) == v[1].get());
}

File: tests/0235_lowest_common_ancestor_of_a_binary_search_tree_test.py
import importlib
from algorithms.tree_node import make_tree

module = importlib.import_module(
    "algorithms.0235_lowest_common_ancestor_of_a_binary_search_tree"
)


def test_lowest_common_ancestor():
    t = make_tree([6, 2, 8, 0, 4, 7, 9, None, None, 3, 5])
    s = module.Solution()
    assert s.lowestCommonAncestor(t[0], t[1], t[2]) == t[0]
    assert s.lowestCommonAncestor(t[0], t[2], t[1]) == t[0]

    assert s.lowestCommonAncestor(t[0], t[1], t[4]) == t[1]
    assert s.lowestCommonAncestor(t[0], t[4], t[1]) == t[1]
